pysheaf.py

```diff
@@ -10,19 +10,20 @@
 import matplotlib.pyplot as plt
 import networkx as nx
 import scipy.optimize
+import copy
 
 ## Data structures
 class Coface: 
     """A coface relation"""
-    def __init__(self,index,orientation):
+    def __init__(self,index=None,orientation=None):
         self.index=index
```

Comment:
This is where I didn't communicate very well. A Coface is a relation between two things. It can't really exist without the two things it relates. I think you should have
```
def __init__(self,face,coface,orientation):
```